### PR TITLE
Hadoop 3.1.0

### PR DESCRIPTION
**Pull request by acmnu**
Following actions was done to build 3.1.0:

Version bump to 3.1.0
Ant version bump
MAPREDUCE-6875 Rename mapred-site.xml.template to mapred-site.xml
HADOOP-13597 There is no kms-config.sh anymore
HADOOP-15122 patch has been removed because it had been merge in 3.1.0
cmake has been replaced with cmake3

### CODE REVIEW

**Comment by acmnu** [comment]
@oflebbe, Please take a look at this.

File: bigtop-packages/src/rpm/hadoop/SPECS/hadoop.spec [hunk 1]
```diff
@@ -623,7 +623,7 @@ fi
 #%config(noreplace) %{etc_hadoop}/conf.empty/mapred-site.xml
 %config(noreplace) %{etc_hadoop}/conf.empty/mapred-env.sh
 %config(noreplace) %{etc_hadoop}/conf.empty/mapred-queues.xml.template
-%config(noreplace) %{etc_hadoop}/conf.empty/mapred-site.xml.template
+%config(noreplace) %{etc_hadoop}/conf.empty/mapred-site.xml
```
Comment (evans-ye) [inline bigtop-packages/src/rpm/hadoop/SPECS/hadoop.spec]: Not sure why remove the template here. Is it because this is also renamed by upstream?

Comment (acmnu) [reply bigtop-packages/src/rpm/hadoop/SPECS/hadoop.spec]: That is upstream change. It was done in https://github.com/apache/hadoop/commit/f1de2c8560a4ce1a346727c2c265570c5b6b872e

Comment (acmnu) [reply bigtop-packages/src/rpm/hadoop/SPECS/hadoop.spec]: Commit and pull request description has been changed to clarify question above.

**Comment by acmnu** [comment]
BTW, I am still in doubt about my solution with cmake3. It needs to bump other projects to understand result.

**Comment by acmnu** [comment]
Seems no one care about it. So I close that one.

**Comment by evans-ye** [comment]
@acmnu sorry got "distracted by work". Did you managed to get it work, or have customer using it?

**Pull request closed without merging**